src/module.cpp

```diff
@@ -476,7 +476,6 @@ instruction_ref module::insert_literal(instruction_ref ins, literal l)
 
 instruction_ref module::insert_parameter(instruction_ref ins, std::string name, shape s)
 {
-    assert(get_parameter_shape(name) == shape{});
```

Comment:
I missed to include this earlier. The assert needed to be removed because we are adding a parameter with the same name as an existing one with a non-default shape. Just wanted to draw your attention to this to confirm that its ok to be removed. @pfultz2

Reply:
We shouldn't remove this assert. We should probably rename the parameter before we insert the new one.

Reply:
I was looking and there does not seem to be existing functionality to change parameter names. Any suggestions on how to approach this?

Reply:
I have added `rename_parameter` function to the module class.